src/StateMachines/StateMachine.php

```diff
@@ -86,6 +85,11 @@ public function hasPendingTransitions()
         return $this->pendingTransitions()->notApplied()->exists();
     }
 
+    public function normalizeEnumCasting($state)
```

Comment:
Even though I know that this PR is explicitly focused on enums, I'd say it might be good naming this method with something more generic (like `normalizeCasting($state)` ). This is just so that more casts can be added in the future with, let's say, custom classes or other defined casts. What do you think about it?

Also, thanks for the PR, I was thinking of adding it myself before checking the existing ones!

Reply:
Thanks for your suggestion, I'll gladly add it.